test/middleware/navigation_middleware/navigation_middleware_test.dart

```diff
@@ -0,0 +1,103 @@
+import 'package:flutter/widgets.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:logotime/middleware/navigation_middleware.dart';
+import 'package:logotime/redux/action/navigation_action.dart';
+import 'package:logotime/redux/action/registration_action.dart';
+import 'package:logotime/redux/action/user/authorization_action.dart';
+import 'package:logotime/redux/state/app/app_state.dart';
+import 'package:logotime/service/navigation/navigation_service.dart';
+import 'package:logotime/service/navigation/route_direction.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+import 'package:redux/redux.dart';
+
+import '../../mock/next_dispatcher_mock.dart';
+import 'navigation_middleware_test.mocks.dart';
+
+@GenerateNiceMocks([
+  MockSpec<INavigationService>(),
+  MockSpec<Store<AppState>>(),
+  MockSpec<Route<dynamic>>(),
```

Comment:
hmm, never used NiceMocks. I did simplified version. JFYI
```
@GenerateMocks([
  INetworkService,
  IDialogService,
  IPermissionService,
  AppStore,
  ShowInfoDialogAction,
])
```

Reply:
I decided to use NiceMocks as it is recommended by the mockito docs:

> Mockito provides two APIs for generating mocks, the @GenerateNiceMocks annotation and the @GenerateMocks annotation.  **The recommended API** is @GenerateNiceMocks. The difference between these two APIs is in the behavior of a generated mock class when a method is called and no stub could be found.